src/lucky/redirectable.cr

```diff
@@ -122,36 +122,20 @@ module Lucky::Redirectable
   def redirect(to path : String, status : Int32 = 302) : Lucky::TextResponse
     # flash messages are not consumed here, so keep them for the next action
     flash.keep
-
-    if ajax? && request.method != "GET"
-      context.response.headers.add "Location", path
-
-      # do not enable form disabled elements for XHR redirects, see https://github.com/rails/rails/pull/31441
-      context.response.headers.add "X-Xhr-Redirect", path
-
-      Lucky::TextResponse.new(context,
-        "text/javascript",
-        %[Turbolinks.clearCache();\nTurbolinks.visit(#{path.to_json}, {"action": "replace"})],
-        status: 200)
-    else
-      if request.headers["Turbolinks-Referrer"]?
-        store_turbolinks_location_in_session(path)
-      end
-      # ordinary redirect
-      context.response.headers.add "Location", path
-      context.response.status_code = status
-      Lucky::TextResponse.new(context, "", "")
-    end
+    special_ajax_redirect(path, status)
   end
 
   # :nodoc:
   def redirect(to page_instead_of_action : Lucky::HTMLPage.class, **unused_args)
     {% raise "You accidentally redirected to a Lucky::HTMLPage instead of a Lucky::Action" %}
   end
 
-  private def store_turbolinks_location_in_session(path : String)
-    cookies.set(:_turbolinks_location, path).http_only(true)
-    # this cookie read at Lucky::RedirectableTurbolinksSupport
+  private def special_ajax_redirect(path, status) : Lucky::TextResponse
```

Comment:
This is really not a special ajax redirect by default. It's only on the turbolinks module that the ✨ added. Maybe rename this and the overrides just `handle_redirect` or something to that effect.

Reply:
Yeah, I wasn't very sure about the naming of this... And now I think to try to only overload a subpart of the `def redirect(to path : String, status : Int32 = 302) : Lucky::TextResponse` might obscure things later on. Dumb code is better than smart code or something :-)

So what do you think about (removing `special_ajax_redirect` and overloading the whole `redirect` method) :

``` Crystal
# In redirectable.cr

  def redirect(to path : String, status : Int32 = 302) : Lucky::TextResponse
    # flash messages are not consumed here, so keep them for the next action
    flash.keep
    context.response.headers.add "Location", path
    context.response.status_code = status
    Lucky::TextResponse.new(context, "", "")
  end
  ```
  And
  
``` Crystal
# In turbolinks_action_support.cr

# Overloads Lucky::Redirectable redirect's method
  def redirect(to path : String, status : Int32 = 302) : Lucky::TextResponse
    # flash messages are not consumed here, so keep them for the next action
    flash.keep
   if ajax? && request.method != "GET"
      context.response.headers.add "Location", path

      # do not enable form disabled elements for XHR redirects, see https://github.com/rails/rails/pull/31441
      context.response.headers.add "X-Xhr-Redirect", path

      Lucky::TextResponse.new(context,
        "text/javascript",
        %[Turbolinks.clearCache();\nTurbolinks.visit(#{path.to_json}, {"action": "replace"})],
        status: 200)
    else
      if request.headers["Turbolinks-Referrer"]?
        store_turbolinks_location_in_session(path)
      end
      # ordinary redirect
      context.response.headers.add "Location", path
      context.response.status_code = status
      Lucky::TextResponse.new(context, "", "")
    end
    context.response.headers.add "Location", path
    context.response.status_code = status
    Lucky::TextResponse.new(context, "", "")
  end

private def store_turbolinks_location_in_session(path : String)
    cookies.set(:_turbolinks_location, path).http_only(true)
  end

  macro included
    before set_turbolinks_location_header_from_session
  end

  private def set_turbolinks_location_header_from_session
    if turbolinks_location = cookies.get?(:_turbolinks_location)
      cookies.delete(:_turbolinks_location)
      # change browser address bar at last request, see https://github.com/turbolinks/turbolinks#following-redirects
      response.headers["Turbolinks-Location"] = turbolinks_location
    end
    continue
  end
  ```

Reply:
I think that's fine 👍